config/queue/simplequeue.go

```diff
@@ -0,0 +1,153 @@
+	retryqueue list.List              // list of queued messages; the list is not multithreading safe, and must only be accessed from backgroundtask()
+}
+
+// Resume informs that the output is working again - can be called multiple times and is thread safe.
+// Should be called after each successfully delivery by the output.
+func (t *simpleQueue) Resume(ctx context.Context) error {
+	if atomic.CompareAndSwapUint32(&t.isInPause, StatusPaused, StatusDelivering) {
+		goglog.Logger.Debug("queue: requesting resume")
+		return t.control.RequestResume(ctx)
+	}
+	return nil
+}
+
+// Queue queues an event into the queue, blocking if necessary until cancelled. Queue is used from the output to put something into the queue.
+// A call to add an event onto the queue will also pause the input.
+func (t *simpleQueue) Queue(ctx context.Context, event logevent.LogEvent) error {
+	if atomic.CompareAndSwapUint32(&t.isInPause, StatusDelivering, StatusPaused) {
+		goglog.Logger.Debug("queue is requesting pause")
+		err := t.control.RequestPause(ctx)
+		if err != nil {
+			goglog.Logger.Error("queue: ", err.Error())
+		}
+	}
+	select {
+	case t.queue <- event:
+		return nil
+	case <-ctx.Done():
+		return ErrContextCancelled.New(nil)
+	}
+}
+
+// NewSimpleQueue returns a new queue using a simple retry/queuing mechanism. receiver is your Output object. queueSize is the number of events that will be queued before dropping events from the queue; with a value of -1 the sky is the limit. retryInterval is the time in seconds between each retry.
+func NewSimpleQueue(ctx context.Context, control config.Control, receiver QueueReceiver, queueSize int, retryInterval uint) Queue {
+	var chanSize int
+	if queueSize < 1 {
+		chanSize = 1
+	} else {
+		chanSize = queueSize
+	}
+	return &simpleQueue{
+		RetryInterval: retryInterval,
+		MaxQueueSize:  queueSize,
+		ctx:           ctx,
+		output:        receiver,
+		control:       control,
+		isInPause:     StatusDelivering,
+		queue:         make(chan logevent.LogEvent, chanSize),
+		retryqueue:    list.List{},
+	}
+}
+
+// GetType satisfies TypeCommonConfig
+func (t *simpleQueue) GetType() string {
+	return t.output.GetType()
+}
+
+// Output satisfies TypeOutputConfig and is handling incoming events
+func (t *simpleQueue) Output(ctx context.Context, event logevent.LogEvent) (err error) {
+	// see if output has requested pause and if so just queue the event instead of trying
+	if atomic.LoadUint32(&t.isInPause) == StatusPaused {
+		select {
+		case t.queue <- event:
+			return nil
+		case <-ctx.Done():
+			return ErrContextCancelled.New(nil)
+		case <-t.ctx.Done():
+			return ErrContextCancelled.New(nil)
+		}
+	}
+	// If we are not in pause mode then call the sender method
+	return t.output.OutputEvent(ctx, event)
+}
+
+// backgroundtask is running in the background and adds new events to the queue, tries to send them out on the RetryInterval.
+func (t *simpleQueue) backgroundtask() {
```

Comment:
https://github.com/tsaikd/gogstash/runs/4495112091

`backgroundtask()` is not triggered.

Reply:
I have solved the issue and added a test case.

Reply:
https://github.com/tsaikd/gogstash/runs/4509922290

additional lint issues

Reply:
I looked into the lint. It seems to be in the test file and because I did not handle an error that was not important for the test. I have posted an update on this.

I did:

funcWithError()

now changed to:

_ = funcWithError()